src/pages/matrix.tsx

```diff
@@ -149,8 +152,37 @@ const MatrixApp = (): JSX.Element => {
   const maxCols = styles?.maxCols;
   const maxRows = styles?.maxRows;
 
-  const rowHeaderWidth = styles?.rowHeaderWidth ? styles?.rowHeaderWidth : 250;
-  const colHeaderHeight = styles?.colHeaderHeight ? styles?.colHeaderHeight : 50;
+  const rowHeaderWidth = styles?.rowHeader?.width ? styles?.rowHeader?.width : 250;
+  const colHeaderHeight = styles?.columnHeader?.width ? styles?.columnHeader?.width : 50;
+
+  // Configuration for headers scrolling
+  let rowHeaderScrollable = styles?.rowHeader?.scrollable ? styles?.rowHeader?.scrollable : false;
```

Comment:
This is very hard to follow. You should add a function to `utils/config.ts` instead that will return the values that you expect. So,

```tsx
export const processMatrixHeaderStyles = (size: number, scrollable?: boolean, scrollableMaxSize?: number) => {
  const validMaxSize = typeof scrollableMaxSize === 'number' && scrollableMaxSize > 0;

  let usedScrollable = !!scrollable;
  let usedMaxSize = scrollableMaxSize;

  if (!validMaxSize || scrollableMaxSize <= size || !usedScrollable) {
    usedMaxSize = size;
    usedScrollable = false;
  }

  if (!validMaxSize && usedScrollable) {
    usedMaxSize = -1;
  }

  return {
    scrollable: usedScrollable,
    scrollableMaxSize: usedMaxSize
  }
}
```

And then in here you just would do,

```tsx
const {
    scrollable: rowHeaderScrollable,
    scrollableMaxSize: rowHeaderMaxWidth
  } = processMatrixHeaderStyles(rowHeaderWidth, styles?.columnHeader?.scrollable, styles?.rowHeader?.scrollableMaxWidth);
```

And similar thing for `colHeader`. Please make sure the function that I wrote covers what you're trying to do. Also make sure you add a proper JSDoc comment to the function and explain what it does.